src/coreclr/vm/runtimehandles.cpp

```diff
@@ -997,6 +929,44 @@ extern "C" void QCALLTYPE QCall_FreeGCHandleForTypeHandle(QCall::TypeHandle pTyp
     END_QCALL;
 }
 
+extern "C" INT32 QCALLTYPE RuntimeTypeHandle_GetNumVirtuals(QCall::TypeHandle pTypeHandle, BOOL includeStaticVirtuals)
+{
+    QCALL_CONTRACT;
+
+    INT32 numVirtuals = 0;
+
+    BEGIN_QCALL;
+
+    TypeHandle typeHandle = pTypeHandle.AsTypeHandle();
+
+    if (typeHandle.IsGenericVariable())
+        COMPlusThrow(kArgumentException, W("Arg_InvalidHandle"));
+
+    MethodTable *pMT = typeHandle.GetMethodTable();
+    if (pMT != NULL)
+    {
+        numVirtuals = (INT32)pMT->GetNumVirtuals();
+
+        if (includeStaticVirtuals
+            && pMT->HasVirtualStaticMethods())
+        {
+            for (MethodTable::MethodIterator it(pMT); it.IsValid(); it.Next())
+            {
+                MethodDesc *pMD = it.GetMethodDesc();
```

Comment:
This is coupled with the logic in `RuntimeTypeHandle::GetMethodAt`. I would be nice to keep the two methods next to each other like they were before.

The two methods together form an iterator used by `RuntimeType.GetInterfaceMap`. It is not an efficient iterator - it has quadratic complexity. There are certainly ways to make it better, I am not sure how much we care.